project_template/BUILDING.md

```diff
@@ -0,0 +1,11 @@
+
+## Requirements
+
+This Azure SDK module has several external build-time requirements.
+
+- **Catch2** (for tests): https://github.com/catchorg/Catch2.git

```

Comment:
>* (for tests): https://github.com/catchorg/Catch2.git [](start = 11, length = 53)

Regarding tools/frameworks that have been used in the past:
The Azure IoTHub SDK team uses the following:

- azure-ctest ([https://github.com/Azure/azure-ctest](https://github.com/Azure/azure-ctest)). This is a very simple and portable test runner for C. At the time of its writing we did consider CppUnitTest as the test runner but the fact that it was C++ and it required small changes for porting even when building with gcc was a no-go for portability.
- azure-c-testrunnerswitcher ([https://github.com/Azure/azure-c-testrunnerswitcher](https://github.com/Azure/azure-c-testrunnerswitcher)). This is a switcher between ctest and CppUnitTest. The reason it exists is because sometimes it is desirable to run, debug analyze tests under VS.
- umock-c ([https://github.com/Azure/umock-c](https://github.com/Azure/umock-c)). This is a C mocking library. It offers recording of calls, comparing expected and actual calls, negative tests, handling various types (charptr, stdint, bool, custom types), allows chaining of modifiers on expectations, etc.

All of the tests in IoTHub C SDK are written with these tools and they have served us well for the past 4-5 years.

Let me know if you find these valuable - I can ramble about this waaaay more :-).